auth/constructor.go

```diff
@@ -73,13 +79,21 @@ func providePrimaryBasculeConstructorOptions(apiBase string) fx.Option {
 				return basculehttp.WithTokenFactory("Basic", basicTokenFactory), nil
 			},
 		},
-
 		fx.Annotated{
-			Group: "primary_bascule_constructor_options",
+			Group: primaryBasculeCOptionsName,
 			Target: func() basculehttp.COption {
 				return basculehttp.WithParseURLFunc(basculehttp.CreateRemovePrefixURLFunc("/"+apiBase+"/", basculehttp.DefaultParseURLFunc))
 			},
 		},
+		fx.Annotated{
+			Group: primaryBasculeCOptionsName,
+			Target: func(in primaryBasculeOnHTTPErrorResponseIn) basculehttp.COption {
+				if in.OnErrorHTTPResponse == nil {
+					return nil
```

Comment:
Won't this add `nil` to the list? I'm not sure that bascule is equipped to handle that?

A solution might be to use the `arrange.IfSet()` function, but I'm not sure.

Reply:
Yeah, nil will be added to the list but it'll be filtered out before using the options here https://github.com/xmidt-org/argus/blob/4761625cca08a0683bb6a464c31e1a71145e2651/auth/constructor.go#L95